packages/provider-elasticsearch/src/example-types/results/highlighting/request.js

```diff
@@ -95,16 +95,19 @@ let getFieldSubFields = (field) =>
 /**
  * Returns object of all subfields in a schema.
  */
-let getSchemaSubFields = (schema) =>
-  F.reduceIndexed(
-    (acc, field, path) =>
-      F.mergeOn(
-        acc,
-        _.mapKeys((k) => `${path}.${k}`, getFieldSubFields(field))
-      ),
-    {},
-    schema.fields
-  )
+let getSchemaSubFields = _.memoize(
+  (schema) =>
+    F.reduceIndexed(
+      (acc, field, path) => {
+        let subFields = getFieldSubFields(field)
+        for (let k in subFields) acc[`${path}.${k}`] = subFields[k]
+        return acc
```

Comment:
Mutating the accumulator and then returning it feels kind of silly to me. Why not just use a `for` loop. It's easy to read and clearly indicates the imperative approach you're taking here.